charms/slurmctld/src/charm.py

```diff
@@ -234,16 +222,17 @@ def _on_write_slurm_conf(
             return
 
         if slurm_config := self._assemble_slurm_conf():
-            self._slurmctld_manager.stop_slurmctld()
-            self._slurmctld_manager.write_slurm_conf(slurm_config)
+            self._slurmctld.service.disable()
+            self._slurmctld.config.dump(slurm_config)
 
             # Write out any user_supplied_cgroup_parameters to /etc/slurm/cgroup.conf.
             if user_supplied_cgroup_parameters := self.config.get("cgroup-parameters", ""):
-                self._slurmctld_manager.write_cgroup_conf(str(user_supplied_cgroup_parameters))
-
-            self._slurmctld_manager.start_slurmctld()
+                self._slurmctld.cgroup.dump(
+                    CgroupConfig.from_str(str(user_supplied_cgroup_parameters))
+                )
```

Comment:
I suppose these have the correct path as a default parameter? It would be nice to have them explicitly passed for the future when we add Snap support on the main branch.

Reply:
The `dump`, `load`, and `edit` methods on the service managers are just light wrappers around the `dump`, `load`, and `edit` functions provided by the different configuration editors supplied by `slurmutils`. The config file path is declared when the service manager is initialized in the charm:

https://github.com/charmed-hpc/hpc-libs/blob/eaa1978d2c51f22e6bffd6a1cea56e54b67534d4/lib/charms/hpc_libs/v0/slurm_ops.py#L911-L921

The config file path will change based on the back-end Slurm implementation being used. If the deb package is being used, the configuration file path will be `/etc/slurm/<*.conf>`, and if the snap implementation is used, the configuration file path will be `/var/snap/slurm/common/etc/slurm/<*.conf>`. So while the paths are not explicitly passed in _charm.py_, the path is changed based on which Slurm implementation we're using.